### PR TITLE
[List][Navigation] Display external URL redirect targets (#1713)

### PR DESCRIPTION
**Pull request by vladbailescu**
* Resolve redirect chain in the link handler

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java [hunk 1]
```diff
@@ -73,6 +99,11 @@
     @OSGiService
     private List<PathProcessor> pathProcessors;
 
+    /**
+     * Variable that defines how
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java:103]: some docs missing here

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java [hunk 2]
```diff
@@ -114,11 +145,20 @@
      */
     @NotNull
     public Optional<Link<Page>> getLink(@Nullable Page page) {
-        if (page == null) {
+        Page resolved = resolveRedirects(page);
+        if (resolved == null && page != null) {
+            String redirectTarget = page.getProperties().get(PageImpl.PN_REDIRECT_TARGET, "");
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java:153]: what about the situation where:
page A -> page B -> external URL
looks like this only handles external redirects if they're at the first level, which is the more common scenario.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java [hunk 3]
```diff
@@ -276,4 +316,51 @@ private String getPageLinkURL(@NotNull Page page) {
         return Optional.ofNullable(pageManager.getPage(path));
     }
 
+    /**
+     * Attempts to resolve the redirect chain starting form the given page, avoiding loops.
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java:323]: starting from the given page

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java [hunk 4]
```diff
@@ -276,4 +316,51 @@ private String getPageLinkURL(@NotNull Page page) {
...
+     *
+     * @param page The starting {@link Page}
+     * @return The {@link Page} the redirect chain resolves too. Can be the original page, if no redirect target is defined or
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java:326]: chain resolves to.

**Comment by richardhand** [comment]
@vladbailescu - this is much cleaner, thanks for adding it. Added a few, mainly minor comments.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java (hunk 1 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java]: Fixed in 06d5f73

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java (hunk 3 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java]: Fixed in 06d5f73

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java (hunk 4 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java]: Fixed in 06d5f73

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java (hunk 2 above)
Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java:164]: I'll prepare a test case and handling for this as well

Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java]: Should be solved by 56f5eab

**Review by richardhand — APPROVED** [review]

**Pull request merged**